scene/gui/flow_container.cpp

```diff
@@ -0,0 +1,249 @@
+			}
+			switch (alignment) {
+				case LINE_ALIGNMENT_TOP:
+					break;
+				case LINE_ALIGNMENT_CENTER:
+					child_rect.position.x += line_data.height / 2 - child_rect.size.width / 2;
+					break;
+				case LINE_ALIGNMENT_BOTTOM:
+					child_rect.position.x += line_data.height - child_rect.size.width;
+					break;
+			}
+		} else { /* HORIZONTAL */
+			if (child->get_v_size_flags() & SIZE_EXPAND) {
+				child_rect.size.height = line_data.height;
+			}
+			switch (alignment) {
+				case LINE_ALIGNMENT_TOP:
+					break;
+				case LINE_ALIGNMENT_CENTER:
+					child_rect.position.y += line_data.height / 2 - child_rect.size.height / 2;
+					break;
+				case LINE_ALIGNMENT_BOTTOM:
+					child_rect.position.y += line_data.height - child_rect.size.height;
+					break;
+			}
+		}
+
+		fit_child_in_rect(child, child_rect);
+		child_idx_in_line++;
+	}
+	cached_height = (vertical ? ofs.x : ofs.y) + line_height;
+	cached_line_count = current_line_idx;
+}
+
+Size2 FlowContainer::get_minimum_size() const {
+	Size2i minimum;
+
+	for (int i = 0; i < get_child_count(); i++) {
+		Control *c = Object::cast_to<Control>(get_child(i));
+		if (!c) {
+			continue;
+		}
+		if (c->is_set_as_top_level()) {
+			continue;
+		}
+
+		if (!c->is_visible()) {
+			continue;
+		}
+
+		Size2i size = c->get_combined_minimum_size();
+
+		if (vertical) {
+			if (size.height > minimum.height) {
+				minimum.height = size.height;
+			}
+
+			minimum.width = cached_height;
+
+		} else {
+			if (size.width > minimum.width) {
+				minimum.width = size.width;
+			}
+
+			minimum.height = cached_height;
+		}
+	}
+
+	return minimum;
+}
+
+void FlowContainer::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_SORT_CHILDREN: {
+			_resort();
+			update_minimum_size();
+		} break;
+		case NOTIFICATION_THEME_CHANGED: {
+			update_minimum_size();
+		} break;
+		case NOTIFICATION_TRANSLATION_CHANGED:
+		case NOTIFICATION_LAYOUT_DIRECTION_CHANGED: {
+			queue_sort();
+		} break;
+	}
+}
+
+int FlowContainer::get_line_count() const {
+	return cached_line_count;
+}
+
+void FlowContainer::set_alignment(LineAlignmentMode p_alignment) {
+	alignment = p_alignment;
+	queue_sort();
+	update_minimum_size();
+}
+
+FlowContainer::LineAlignmentMode FlowContainer::get_alignment() const {
+	return alignment;
+}
+
+FlowContainer::FlowContainer(bool p_vertical) {
+	vertical = p_vertical;
+}
+
+void FlowContainer::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("get_alignment"), &FlowContainer::get_alignment);
+	ClassDB::bind_method(D_METHOD("set_alignment", "alignment"), &FlowContainer::set_alignment);
+
+	ClassDB::bind_method(D_METHOD("get_line_count"), &FlowContainer::get_line_count);
+
+	BIND_ENUM_CONSTANT(LINE_ALIGNMENT_TOP);
+	BIND_ENUM_CONSTANT(LINE_ALIGNMENT_CENTER);
+	BIND_ENUM_CONSTANT(LINE_ALIGNMENT_BOTTOM);
+
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "In-line alignment", PROPERTY_HINT_ENUM, "Top,Center,Bottom"), "set_alignment", "get_alignment");
```

Comment:
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::INT, "alignment", PROPERTY_HINT_ENUM, "Top,Center,Bottom"), "set_alignment", "get_alignment");
```

That's property name not a description, should not have any spaces or `-`. That's probably causing Mono build to fail.

Reply:
Fixed.